R/module_transform_data.R

```diff
@@ -9,24 +9,32 @@
 #'
 #'
 #' @name module_transform_data
-#' @keywords internal
 NULL
 
+#' @export
 #' @rdname module_transform_data
-ui_transform_data <- function(id, transformers = list(), class = "well") {
+ui_teal_transform_data <- function(id, transformators, class = "well") {
   checkmate::assert_string(id)
-  checkmate::assert_list(transformers, "teal_transform_module")
+  if (length(transformators) == 0L) {
+    return(NULL)
+  }
+  if (inherits(transformators, "teal_transform_module")) {
+    transformators <- list(transformators)
+  }
+  checkmate::assert_list(transformators, "teal_transform_module")
 
-  ns <- NS(id)
-  labels <- lapply(transformers, function(x) attr(x, "label"))
+  labels <- lapply(transformators, function(x) attr(x, "label"))
   ids <- get_unique_labels(labels)
```

Comment:
I'm not a big fan of using labels as ID. We have made a few spectacular failure becuase of that. Labels can contain special characters and can be not unique. I would rather use label as label and ID to be "<prefix>_<index>", e.g. here: "wrapper_1", "wrapper_2", ...

Reply:
Me neither, I thought for a quick second if we should use the same approach as `teal.slice` by appending an hash to the end to a sanitized id.

But it's unnecessary, `transformator_x` should be better on  the long term.

Reply:
a71e6d560